src/Twilio/Security/RequestValidator.php

```diff
@@ -130,4 +136,50 @@ public static function compare($a, $b) {
 
     }
 
+    /*
+     * Removes the port from the URL
+     * @param $parsedURL array
+     * @returns string Full URL without the port number
+     */
+    private static function removePort($parsedUrl) {
+        unset($parsedUrl['port']);
+        return RequestValidator::buildUrl($parsedUrl);
+    }
+
+    /*
+     * Adds the port to the URL
+     * @param $parsedURL array
+     * @returns string Full URL with the port number
+     */
+    private static function addPort($parsedUrl) {
+        if (!isset($parsedUrl['port'])) {
+            $port = ($parsedUrl['scheme'] === 'https') ? 443 : 80;
+            $parsedUrl['port'] = $port;
+        }
+        return RequestValidator::buildUrl($parsedUrl);
+    }
+
+    /*
+     * Builds the URL from its parsed component pieces
+     * @param $parsedURL array
+     * @returns string Full URL
+     */
+    private static function buildUrl($parsedUrl) {
+        $url = '';
+        $parts = array();
+
+        $parts['scheme'] = isset($parsedUrl['scheme']) ? $parsedUrl['scheme'] . '://' : '';
+        $parts['host'] = isset($parsedUrl['host']) ? $parsedUrl['host'] : '';
+        $parts['port'] = isset($parsedUrl['port']) ? ':' . $parsedUrl['port'] : '';
+        $parts['user'] = isset($parsedUrl['user']) ? $parsedUrl['user'] : '';
+        $parts['pass'] = isset($parsedUrl['pass']) ? ':' . $parsedUrl['pass'] : '';
+        $parts['pass'] = ($parts['user'] || $parts['pass']) ? $parts['pass'] . '@' : '';
+        $parts['path'] = isset($parsedUrl['path']) ? $parsedUrl['path'] : '';
+        $parts['query'] = isset($parsedUrl['query']) ? '?' . $parsedUrl['query'] : '';
+        $parts['fragment'] = isset($parsedUrl['fragment']) ? '#' . $parsedUrl['fragment'] : '';
+
+        return \implode('', $parts);
```

Comment:
Not convinced this works properly when the URL has creds. Assume it's adapted from this: https://www.php.net/manual/en/function.parse-url.php#106731

More tests would convince me.

Reply:
You're right, builder adds creds to the wrong location. New tests added to validate creds in URL.